core/new-gui/src/app/workspace/component/navigation/navigation.component.html

```diff
@@ -94,18 +94,22 @@
         <button (click)="tourService.toggle()" class="animate-to-reveal-stop-button" nz-button nzType="default">
           <i nz-icon nzType="question-circle" nzTheme="outline"></i>
         </button>
-        <button nz-button (click)="onClickRunHandler()" [disabled]="runDisable"
+        <button nz-button #runStopButton (click)="handleKill()"
+                [disabled]="executionState == ExecutionState.Uninitialized || executionState == ExecutionState.Completed || executionState == ExecutionState.Failed"
```

Comment:
It is not a good way to bind function call in angular template because of how change detection works. Therefore for input bindings, we should always bind to variables or statements like this, but not funciton calls. 

See this post: https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496

Reply:
good to know!